### ISSUE TITLE
shadow on propeller (fgfs next as of 19.11.2021)

### ISSUE THREAD

**Issue by hbeni** [HHS81/c182s#470]
On yesterday next I got this by chance.
In the propeller I have seen the interiour shadow of the plane. Is that intentional?
Sun was in my back, fair weather setting, afternoon.

![grafik](https://user-images.githubusercontent.com/13608602/146695481-0805d571-8e1a-4942-810f-ae59e5b23c3d.png)

----
_Steps to reproduce:_
- Fair weather
- Time=Afternoon (noon works too)
- Start in air (approach or so, and use the cruising preset)
- Fly so, that the sun casts a shadow to the cockpit
- Try to move the shadow onto the disc and observe shadow casts there

----
SF FGFS Issue ticket: https://sourceforge.net/p/flightgear/codetickets/2713/

**Comment by HHS81** [comment]
that`s new behavior and not intended. Probably a graphic glitch some where in the hardcode.

**Comment by hbeni** [comment]
Ok, I think I remember others having similar problems and it’s been worked on. There was some traffic in the forum and mailing list i think: 
https://forum.flightgear.org/viewtopic.php?f=37&t=39632&hilit=Clipping#p391662

there it was the DDS cache corrupted.

**Comment by hbeni** [comment]
I mark this as WONTFIX for now, but keep it open, so we don't forget that we test once in a while...

**Comment by hbeni** [comment]
C172p has the same issue.

**Comment by hbeni** [comment]
asked the mailing list.

**Comment by hbeni** [comment]
Opened an FGFS issue ticket: https://sourceforge.net/p/flightgear/codetickets/2713/

**Comment by hbeni** [comment]
asked again on the list....

**Comment by hbeni** [comment]
> A quick searching: model-interior-ALS defines a 'shadow-mapping-enabled' flag, which eg the C172 sets in a few places, eg: ./Aircraft/c172p/Models/Effects/interior/c172p-interior.eff: <shadow-mapping-enabled>true</shadow-mapping-enabled>
>
>And the compositor shadows use the uniform 'shadows_enabled' which is controlled by modef-default.eff,
>
>```
><shadows_enabled>
><use>/sim/rendering/shadows/enabled</use>
></shadows_enabled>
>```
>
>What you might need is to subclacc model-default.eff and make a model-unshadowd.eff for your prop disc? I don't know if this is something which makes sense to include in FGData effects.

@TheFGFSEagle Hi Eagle, does that make sense to you? Do you have some spare time to assist?

**Comment by TheFGFSEagle** [comment]
Yeah, I think too what you're seeing on the prop is not the shadow from the shadow map from the interior effects, but simply the compositor shadow - in which case James' idea should work fine.

**Comment by hbeni** [comment]
Hm what still puzzles me a bit - when we have no shadow on the fast propeller disc, where does the light come from, when the sun is behind the plane.
Shouldn't there be a shadow in the disc, even for fast rotating blades?

I think we probably need to do real life investigation.

**Comment by hbeni** [comment]
Asked on the forums: https://forum.flightgear.org/viewtopic.php?f=34&t=41161

**Comment by TheFGFSEagle** [comment]
IMO the shadow should on the prop disc too, but faded by the same amount as the prop - which is just what we have right now, right ?

**Comment by hbeni** [comment]
So - I decided to merge Eagles fix.
The reasoning is that this restores the years-old previous behaviour which sounds more sensible currently.

The shadowing was definitely wrong, and if we get better details on how a real shadow would look like, we now can model that properly trough the new separated effect.

**Comment by hbeni** [comment]
Looks like its again here....
Tried by spawning on Keflavik RW02, Noon time, so the sun is in the back. Idling on the threshold, the shadow is to be seen on the disc.

**Comment by TheFGFSEagle** [comment]
Hmmmmm, that's weird … I think I should be able to take a look later in the afternoon / evening.

**Comment by TheFGFSEagle** [comment]
With FGFS from 17.10., it is still working - just noting that here to facilitate the search for the offending commit.

**Comment by hbeni** [comment]
Still occuring in 2024.1 (release candidate)
